### DISCUSSION TITLE
merge title bar and menu bar

### DISCUSSION THREAD

**Discussion by WangYi13** [godotengine/godot-proposals#4053] [GUI]
just want to save some space.

**Comment by rakkarage** [comment]
[![Untitled.png](https://i.postimg.cc/rs9nZXST/Untitled.png)](https://postimg.cc/XBpkXPH1)

**Comment by Calinou** [comment]
This requires implementing client-side decorations, which is a lot of work to do correctly. See also https://github.com/godotengine/godot-proposals/issues/1868 which also depends on client-side decorations.

**Reply by ghost** [reply]
Let's not shy away from work, dude!

**Reply by Zireael07** [reply]
Did you read the code of conduct? This is very much not a constructive comment... not to mention Godot's fairly stretched when it comes to available contributors who would be willing to do such complex work (smaller features get contributors quicker but this is not only huge but also requires knowledge of OS specific things, and Godot usually only accepts features that work on all supported OSes, ie. Windows, Linux, Mac and Android)

**Reply by ghost** [reply]
If it's not a constructive comment, then why did you write it?

**Reply by Calinou** [reply]
@Luke-devo Please stop baiting other people. This is your last warning – future violations will lead to a ban.

**Reply by ghost** [reply]
What are you talking about?

**Comment by mieldepoche** [comment]
duplicate of #4648
